lib/Core/Site/QueryType/Base.php

```diff
@@ -394,18 +401,36 @@ private function resolveFilterCriteria(array $parameters): ?Criterion
      */
     private function getSortClauses(array $parameters): array
     {
+        $sortDefinitions = $parameters['sort'];
+        $sortClauses = [];
+
+        if (empty($sortDefinitions)) {
+            return $sortClauses;
+        }
+
+        if (\strtolower(\trim($sortDefinitions)) === 'parent' && array_key_exists('location', $parameters)) {
```

Comment:
Two issues with handling parent sort clause like this:

- `sort: parent` is in fact an alias for `sort: [parent]`, this makes combining with other sort clauses impossible
- will not give an error on incorrect usage (in Content query type)

I would propose to solve these by implementing it in the `SortClauseParser` service. If the Location can't be found, invalid argument exception should be thrown, and the existing mechanism will take care of throwing the exception in `self::parseCustomSortString()`.

Alternatively, it's possible to configure sort with `SortClause` instances, so we can also handle it through an expression:

```yaml
sort: '@=location.innerLocation.getSortClauses()'
```

Reply:
@pspanja Sorry, I didn't know that it's possible to provide directly sort clauses. I reverted all the changes and I will use it with direct sort clauses.